test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -1064,4 +1073,466 @@ defmodule Trento.SapSystemTest do
+        [
+          build(
+            :database_registered_event,
+            sap_system_id: sap_system_id
+          ),
+          build(
+            :database_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: database_host_id,
+            instance_number: database_instance_number
+          ),
+          build(
+            :application_instance_registered_event,
+            sap_system_id: sap_system_id,
+            features: "MESSAGESERVER|ENQUE",
+            host_id: message_server_host_id,
+            instance_number: message_server_instance_number
+          ),
+          build(
+            :application_instance_registered_event,
+            sap_system_id: sap_system_id,
+            features: "ABAP|GATEWAY|ICMAN|IGS",
+            host_id: abap_host_id,
+            instance_number: abap_instance_number
+          ),
+          build(
+            :sap_system_registered_event,
+            sap_system_id: sap_system_id
+          ),
+          build(
+            :application_instance_registered_event,
+            sap_system_id: sap_system_id,
+            host_id: enqrep_host_id,
+            instance_number: enqrep_server_instance_number,
+            features: "ENQREP"
+          )
+        ],
+        %DeregisterApplicationInstance{
+          sap_system_id: sap_system_id,
+          host_id: abap_host_id,
+          instance_number: abap_instance_number,
+          deregistered_at: deregistered_at
+        },
+        [
+          %ApplicationInstanceDeregistered{
+            sap_system_id: sap_system_id,
+            host_id: abap_host_id,
+            instance_number: abap_instance_number,
+            deregistered_at: deregistered_at
+          },
+          %SapSystemDeregistered{
+            sap_system_id: sap_system_id,
+            deregistered_at: deregistered_at
+          }
+        ],
+        fn sap_system ->
+          assert %SapSystem{
+                   database: %Database{
+                     instances: [
+                       %Instance{
+                         instance_number: ^database_instance_number,
+                         host_id: ^database_host_id
+                       }
+                     ]
+                   },
+                   application: %Application{
+                     instances: [
+                       %Instance{
+                         instance_number: ^enqrep_server_instance_number,
+                         host_id: ^enqrep_host_id
+                       },
+                       %Instance{
+                         instance_number: ^message_server_instance_number,
+                         host_id: ^message_server_host_id
+                       }
+                     ]
+                   },
+                   deregistered_at: ^deregistered_at
+                 } = sap_system
+        end
+      )
+    end
+
+    test "should deregister just a Message Server from a not-fully-registered SAP system" do
```

Comment:
I would remove this test, it does exactly the same than the first. There is no any distinction in reality in the code (we know that elixis'r pattern matching works)

Reply:
We checked this code path/scenario for coverage using _Coveralls_; this test covers the scenario where a SAP System is **not** registered, and _then_ an instance is deregistered - testing that a `SapSystemDeregistered` event is **not** emitted in this case. This is not the same as the first test - removing an ENQREP server, as it would never emit a `SapSystemDeregistered` event.